File: rabbit-mq/hello-world/sender.py
#!/usr/bin/env python
import pika
from time import sleep


def main():
    # Establish a connection to the RabbitMQ Server
    connection = pika.BlockingConnection(
        pika.ConnectionParameters('localhost'))
    channel = connection.channel()

    # Check that the recipent queue exists before sending a message
    # If it does not exists the message will be dropped
    channel.queue_declare(queue='hello')

    # Now we want to send a message to the 'hello' queue
    # In RabbitMQ a message can never be sent directly to the queue, it always needs to go through an exchange.
    message_sent = 'Hello World!'
    channel.basic_publish(exchange='',
                          routing_key='hello',
                          body=message_sent)
    print(" [x] Sent '%s'" % message_sent)

    connection.close()


if __name__ == '__main__':
    NUM_WORKS = 3
    for i in range(NUM_WORKS):
        main()
        sleep(1)

File: rabbit-mq/hello-world/sender.jl
using AMQPClient


function main()
    # Establish connection to RabbitMQ Server
    port = AMQPClient.AMQP_DEFAULT_PORT
    login = "guest"  # default is usually "guest"
    password = "guest"  # default is usually "guest"
    auth_params = Dict{String,Any}("MECHANISM"=>"AMQPLAIN", "LOGIN"=>login, "PASSWORD"=>password)

    conn = connection(; virtualhost="/", host="localhost", port=port, auth_params=auth_params, amqps=nothing)

    chan = channel(conn, AMQPClient.UNUSED_CHANNEL, true)

    success, message_count, consumer_count = queue_declare(chan, "hello")
    
    EXCG_DIRECT = ""
    message = "Hello World!"
    M = Message(Vector{UInt8}(message), content_type="text/plain") # , delivery_mode=PERSISTENT
    basic_publish(chan, M; exchange=EXCG_DIRECT, routing_key="hello")
    
    println(" [x] Sent '$message'")

    if isopen(conn)
        close(conn)
        # close is an asynchronous operation. To wait for the negotiation to complete:
        AMQPClient.wait_for_state(conn, AMQPClient.CONN_STATE_CLOSED)
    end
end

main()